assets/js/frontend/give-donations.js

```diff
@@ -162,14 +181,76 @@ Give = {
 		 * @return {string}
 		 */
 		getGlobalVar: function (str) {
-			if ('undefined' === give_global_vars[str]) {
+			if ('undefined' === typeof give_global_vars[str]) {
 				return '';
 			}
 
 			return give_global_vars[str];
+		},
+
+		/**
+		 * set cache
+		 *
+		 * @since 1.8.17
+		 *
+		 * @param {string} key
+		 * @param {string} value
+		 * @param {object} $form
+		 */
+		setCache: function (key, value, $form) {
+			if ($form.length) {
+				Give.cache['form_' + Give.form.fn.getInfo('form-id', $form)][key] = value;
+			} else {
+				Give.cache[key] = value;
+			}
+		},
+
+		/**
+		 * Get cache
+		 *
+		 * @since 1.8.17
+		 * @param key
+		 * @param $form
+		 * @return {string|*}
+		 */
+		getCache: function (key, $form) {
+			var cache;
+
+			if ($form.length) {
+				cache = Give.cache['form_' + Give.form.fn.getInfo('form-id', $form)][key];
+			} else {
+				cache = Give.cache[key];
+			}
+
+			cache = 'undefined' === typeof cache ? '' : cache;
+
+			return cache;
+		},
+
+		/**
+		 * Initialize cache.
+		 *
+		 * @since 1.8.17
+		 * @private
+		 */
+		__initialize_cache: function () {
```

Comment:
@DevinWalker this class will initialise empty cache for each donation form on current page